### PR TITLE
Upgrades and stuff

### PR DESCRIPTION
**Pull request by epage**
Dependencies
- Upgrade dependencies
- Remove reliance on rustc_serialize

Otherwise, I moved Assert into its own module in prep for future refactorings.

### CODE REVIEW

**Comment by epage** [comment]
This includes #34, will rebase once that is merged.

File: Cargo.toml [hunk 1]
```diff
@@ -13,16 +13,16 @@ keywords = ["cli", "testing", "assert"]
...
+error-chain = "0.11"
+json = "0.11"
 
 [build-dependencies]
-skeptic = "0.5"
+skeptic = "0.13"
 
 [dev-dependencies]
-skeptic = "0.5"
+skeptic = "0.13"
```
Comment (killercup) [inline Cargo.toml]: Wow, that's a big jump from 0.5 to 0.13! :)

File: Cargo.toml [hunk 2]
```diff
@@ -16,7 +16,7 @@ build = "build.rs"
 colored = "1.5"
 difference = "1.0"
 error-chain = "0.11"
-rustc-serialize = "0.3"
+json = "0.11"
```
Comment (killercup) [inline Cargo.toml]: I'd usually prefer to use serde_json everywhere, but it's fine to go with this crate here as it does what we need it to do and also has no dependencies.

Comment (epage) [reply Cargo.toml]: I went with it for that and because I thought `serde_json` was implemented in terms of it which I just found out it isn't.

Comment (epage) [reply Cargo.toml]: I've gone ahead and switched it to `serde_json` (already shows up in the dependency tree anyways).

**Comment by epage** [comment]
I wonder if I can do it

bors r+

**Comment by epage** [comment]
Ah, I take it bors auto-merges

**Comment by killercup** [comment]
I added you as a collaborator on github and with bors! In the meantime, let's give this another try:

bors r+

**Pull request merged**